### PR TITLE
Use op2ext for logging

### PR DESCRIPTION
**Pull request by Brett208**
Plan to remove other Log functions in similar faction over time.

### CODE REVIEW

File: Log.cpp [hunk 1]
```diff
@@ -89,9 +91,9 @@ std::string FormatPacket(const OP2Internal::Packet& packet)
 }
 
 
-void Log(const char* string)
+void Log(const std::string& message)
```
Comment (DanRStevens) [inline Log.cpp:94]: Probably slightly better to split this change into a separate commit. I like it though.

File: OPUNetTransportLayer.cpp [hunk 2]
```diff
@@ -302,9 +300,8 @@ bool OPUNetTransportLayer::JoinGame(HostedGameInfo &game, const char* password)
 	strncpy_s(packet.tlMessage.joinRequest.password, password, sizeof(packet.tlMessage.joinRequest.password));
 
 // **DEBUG**
-logFile << "Sending join request: ";
-LogAddress(game.address);
-logFile << std::endl << "  Session ID: ";
+	Log("Sending join request: " + FormatAddress(game.address));
```
Comment (DanRStevens) [inline OPUNetTransportLayer.cpp]: The indentation looks off here.

File: OPUNetTransportLayer.cpp [hunk 3]
```diff
@@ -1128,9 +1122,8 @@ bool OPUNetTransportLayer::DoImmediateProcessing(Packet &packet, sockaddr_in &fr
 			{
 				tlMessage.tlHeader.commandType = tlcJoinGranted;
 				// **DEBUG**
-				logFile << "Client join accepted: ";
-				LogAddress(from);
-				logFile << " (" << tlMessage.joinReply.newPlayerNetID << ")" << std::endl;
+				Log("Client join accepted: " + FormatAddress(from) + 
+					" (" + std::to_string(tlMessage.joinReply.newPlayerNetID) + ")");
```
Comment (DanRStevens) [inline OPUNetTransportLayer.cpp:1126]: Could potentially change this to use the `FormatPlayerNetID`. I understand why you'd want to do that separately though, as that would change the output slightly.

File: OPUNetTransportLayer.cpp [hunk 4]
```diff
@@ -1413,9 +1404,11 @@ void OPUNetTransportLayer::CheckSourcePort(Packet &packet, sockaddr_in &from)
...
 		{
 			// Port mismatch. Issue warning
-			logFile << "Packet from player " << sourcePlayerIndex << " (";
-			LogAddress(from);
-			logFile << ") received on unexpected port (" << ntohs(sourcePort) << " instead of " << ntohs(expectedPort) << ")  [PlayerNetId: " << sourcePlayerNetId << "]" << std::endl;
+			Log("Packet from player " + std::to_string(sourcePlayerIndex) + 
+				" (" + FormatAddress(from) + ") received on unexpected port (" + 
+				std::to_string(ntohs(sourcePort)) + " instead of " + 
+				std::to_string(ntohs(expectedPort)) + ")  [PlayerNetId: " + 
+				std::to_string(sourcePlayerNetId) + "]");
```
Comment (DanRStevens) [inline OPUNetTransportLayer.cpp:1411]: This can eventually become `FormatPlayerNetID`. I was likely too lazy when I wrote this to break up the line and make a separate call to output a formatted value. The new methods are much more convenient to use.

**Review by DanRStevens — APPROVED** [review]
Excellent. This makes the logging code much less awful.

**Comment by Brett208** [comment]
I was sort of not touching the indent/debug comments. This may mark the code that we want to restrict to some sort of verbose logging eventually.

File: OPUNetTransportLayer.cpp (hunk 4 above)
Comment (Brett208) [reply OPUNetTransportLayer.cpp:1411]: I'll push an issue for future incorporation

**Comment by Brett208** [comment]
Thanks for the quick review on the code.

-Brett
